utils/fields.py

```diff
@@ -16,6 +16,7 @@ class BaseAPIField:
     name: str = ""
     required: bool = False
     discriminator_values: list = None
+    definition: dict = None
```

Comment:
HmmmMMM, I would pick apart the definition here. The idea is to be independent from the YAML at this level. I would try to be explicit instead:
```python
minimum: int = None  # unless we want default values here
maximum: int = None
```
And then something like this in `generate_fake_data`:
```python
   params = [
       minimum if minimum else 0,
       maximum if maximum else 0,  # what should this else be? I don't remember how randint handles no max
   ]
   
    return faker.get_fake_data_for_type(self.data_type, params if params else [])
```

Then you would need to change the `BaseAPIField` initialization to something like:
```python
return BaseAPIField(data_type=data_type, name=name, minimum=definition.get("minimum") ... etc
```

You could also subclass BaseAPIField into `IntegerField` or something as well. But the idea is to be definition-agnostic in the fields level.

Reply:
Yea this is primarily what I needed your eyes on, I don't know the parser inside and out so this seems like the better abstraction level.